hibernate-reactive-core/src/main/java/org/hibernate/reactive/engine/jdbc/mutation/internal/ReactiveMutationExecutorPostInsert.java

```diff
@@ -21,12 +35,105 @@ public ReactiveMutationExecutorPostInsert(
+						getJdbcValueBindings(),
+						modelReference,
+						session
+				)
+				.thenApply( this::logId )
+				.thenCompose(id -> {
+					if (secondaryTablesStatementGroup == null) {
+						return completedFuture(id);
+					}
+					AtomicReference<CompletionStage<Object>>  res = new AtomicReference<>(completedFuture(id));
+					secondaryTablesStatementGroup.forEachStatement((tableName, statementDetails) -> {
+						res.set(res.get().thenCompose(i -> reactiveExecuteWithId(i, tableName, statementDetails, inclusionChecker, resultChecker, session)));
+					});
+					return res.get();
+				});
+	}
+
+	private Object logId(Object identifier) {
+		if ( MODEL_MUTATION_LOGGER_TRACE_ENABLED ) {
+			MODEL_MUTATION_LOGGER
+					.tracef( "Post-insert generated value : `%s` (%s)", identifier, mutationTarget.getNavigableRole().getFullPath() );
+		}
+		return identifier;
+	}
+
+	private CompletionStage<Object> reactiveExecuteWithId(
+			Object id,
+			String tableName,
+			PreparedStatementDetails statementDetails,
 			TableInclusionChecker inclusionChecker,
 			OperationResultChecker resultChecker,
 			SharedSessionContractImplementor session) {
-		throw LOG.nonReactiveMethodCall( "executeReactive" );
+
+		if ( statementDetails == null ) {
+			return completedFuture(id);
+		}
+
+		final EntityTableMapping tableDetails = (EntityTableMapping) statementDetails.getMutatingTableDetails();
+		assert !tableDetails.isIdentifierTable();
+
+		if ( inclusionChecker != null && !inclusionChecker.include( tableDetails ) ) {
+			if ( MODEL_MUTATION_LOGGER_TRACE_ENABLED ) {
+				MODEL_MUTATION_LOGGER.tracef(
+						"Skipping execution of secondary insert : %s",
+						tableDetails.getTableName()
+				);
+			}
+			return completedFuture(id);
+		}
+
+
+		tableDetails.getKeyMapping().breakDownKeyJdbcValues(
+				id,
+				(jdbcValue, columnMapping) -> {
+					valueBindings.bindValue(
+							jdbcValue,
+							tableName,
+							columnMapping.getColumnName(),
+							ParameterUsage.SET
+					);
+				},
+				session
+		);
+
+		session.getJdbcServices().getSqlStatementLogger().logStatement( statementDetails.getSqlString() );
+
+		Object[] params = PreparedStatementAdaptor.bind( statement -> {
+			PreparedStatementDetails details = new PrepareStatementDetailsAdaptor( statementDetails, statement, session.getJdbcServices() );
+			//noinspection resource
+			details.resolveStatement();
+			valueBindings.beforeStatement( details );
+		} );
+
+		ReactiveConnection reactiveConnection = ( (ReactiveConnectionSupplier) session ).getReactiveConnection();
+		String sqlString = statementDetails.getSqlString();
+		return reactiveConnection
+				.update( sqlString, params )
+				.thenCompose( affectedRowCount -> {
+					if ( affectedRowCount == 0 && tableDetails.isOptional() ) {
+						// the optional table did not have a row
+						return completedFuture(id);
+					}
+					checkResults( session, statementDetails, resultChecker, affectedRowCount, -1);
+					return completedFuture(id);
```

Comment:
You can use `.thenApply`, instead of `.thenCompose`,  and just return the `id`

Reply:
Good catch! Fixed